test/blackbox-tests/test-cases/reporting-of-cycles.t/run.t

```diff
@@ -9,20 +9,27 @@ the second run of dune.
      alias a/.a-files
   -> alias b/.b-files
   -> alias a/.a-files
+  -> required by alias a/.a-files
+  -> required by alias package-cycle in dune:1
   [1]
 
   $ dune build @simple-repro-case
   Error: Dependency cycle between:
      _build/default/x
   -> _build/default/y
   -> _build/default/x
+  -> required by _build/default/y
+  -> required by alias simple-repro-case in dune:5
   [1]
 
   $ dune build x1
   Error: Dependency cycle between:
      _build/default/x2
   -> _build/default/x3
   -> _build/default/x2
+  -> required by _build/default/x3
+  -> required by _build/default/x2
+  -> required by _build/default/x1
```

Comment:
I wonder if it's possible to shorten such cycle error messages, like this:

```
     _build/default/x2
  -> _build/default/x3
  -> _build/default/x2
  -> required by _build/default/x1
```

That is, remove the part of the path to the cycle that overlaps with the cycle.

Reply:
That's a good idea, I'll give that a shot.

Reply:
I pushed a commit e640bd6a3255eebf7d17ccc2d8028f474e978453 that implements this. It does indeed shorten a lot of cycles, and makes the messages less confusing.

In that commit I also:
- Changed it so that we report low-level memo stack for code errors. This is something that I wanted in the past and needed when debugging this patch.
- Changed the payload of Cycle_error to be a raw cycle with no duplication between the first and the last element of the list. (a change to the cycle detection library)

Reply:
Nice, much better this way!